src/du_parameters/frequency.py

```diff
@@ -0,0 +1,649 @@
+             False otherwise.
+        """
+        if isinstance(other, GSCN):
+            return self._channel <= other._channel
+        if isinstance(other, int | Decimal):
+            return self._channel <= other
+        return False
+
+    def __ge__(self, other: Any) -> bool:
+        """Check if the current GSCN instance is greater than or equal to others.
+
+        Args:
+            other (Any): The value to compare.
+
+        Returns:
+            bool: True if the current GSCN instance is greater than or equal to the other,
+             False otherwise.
+        """
+        if isinstance(other, GSCN):
+            return self._channel >= other._channel
+        if isinstance(other, int | Decimal):
+            return self._channel >= other
+        return False
+
+    def __sub__(self, other: Any) -> "GSCN":
+        """Subtract another GSCN or integer or decimal from this GSCN.
+
+        Args:
+            other (Any): The value to subtract.
+
+        Returns:
+            GSCN: A new GSCN instance with the updated channel.
+
+        Raises:
+            NotImplementedError: If the other value is not an GSCN, int or Decimal.
+        """
+        if isinstance(other, GSCN):
+            return GSCN(self._channel - other._channel)  # type: ignore[operator]
+        if isinstance(other, int | Decimal):
+            return GSCN(self._channel - round(other))
+        raise NotImplementedError(f"Unsupported type for subtraction: {type(other).__name__}")
+
+    def __add__(self, other: Any) -> "GSCN":
+        """Add another GSCN or integer or Decimal to this GSCN.
+
+        Args:
+            other (Any): The value to add.
+
+        Returns:
+            GSCN: A new GSCN instance with the updated channel.
+
+        Raises:
+            NotImplementedError: If the other value is not an GSCN, int or Decimal.
+        """
+        if isinstance(other, GSCN):
+            return GSCN(self._channel + other._channel)  # type: ignore[operator]
+        if isinstance(other, int | Decimal):
+            return GSCN(self._channel + round(other))
+        raise NotImplementedError(f"Unsupported type for addition: {type(other).__name__}")
+
+    def __truediv__(self, other: Any) -> "GSCN":
+        """Divide GSCN to other GSCN, integer or Decimal.
+
+        Args:
+            other (Any): The value to divide by.
+
+        Returns:
+            GSCN: A new GSCN instance with the updated channel.
+
+        Raises:
+            NotImplementedError: If the other value is not an GSCN, int or Decimal.
+        """
+        if isinstance(other, GSCN):
+            return GSCN(round(self._channel / other._channel))  # type: ignore[operator]
+        if isinstance(other, int | Decimal):
+            return GSCN(round(self._channel / other))  # type: ignore[operator]
+        raise NotImplementedError(f"Unsupported type for division: {type(other).__name__}")
+
+    @classmethod
```

Comment:
I'm really not sure about this being a `classmethod`. What we're doing here is turning the GSCN channel to frequency; and GSCN channel belongs to the instance of the GSCN class not the class itself.
Shouldn't the method signature simply be `def to_frequency(self) -> Frequency`?

Reply:
BTW, with the above we could also avoid awkward constructions like:
```python
freq = GSCN.to_frequency(GSCN(gscn))
```
IMO 
```python
freq = GSCN(gscn).to_frequency()
```` 
looks much better.

Similarly in the `absolute_freq_ssb.py`:
```python
gscn = GSCN.from_frequency(center_freq)
adjusted_frequency = GSCN.to_frequency(gscn)
```
could turn into nicer:
```python
gscn = GSCN.from_frequency(center_freq)
adjusted_frequency = gscn.to_frequency()
```
WDYT @ghislainbourgeois